pkg/process/procutil/process_model.go

```diff
@@ -33,13 +62,55 @@ type Stats struct {
 	Nice        int32
 	OpenFdCount int32
 	NumThreads  int32
+	CPUPercent  *CPUPercentStat
 	CPUTime     *CPUTimesStat
 	MemInfo     *MemoryInfoStat
 	MemInfoEx   *MemoryInfoExStat
 	IOStat      *IOCountersStat
+	IORateStat  *IOCountersRateStat
 	CtxSwitches *NumCtxSwitchesStat
 }
 
+// DeepCopy creates a deep copy of Stats
+func (s *Stats) DeepCopy() *Stats {
+	copy := &Stats{
+		CreateTime:  s.CreateTime,
+		Status:      s.Status,
+		Nice:        s.Nice,
+		OpenFdCount: s.OpenFdCount,
+		NumThreads:  s.NumThreads,
+	}
+	if s.CPUTime != nil {
+		copy.CPUTime = &CPUTimesStat{}
+		*copy.CPUTime = *s.CPUTime
+	}
+	if s.CPUPercent != nil {
+		copy.CPUPercent = &CPUPercentStat{}
+		*copy.CPUPercent = *s.CPUPercent
+	}
+	if s.MemInfo != nil {
+		copy.MemInfo = &MemoryInfoStat{}
+		*copy.MemInfo = *s.MemInfo
+	}
+	if s.MemInfoEx != nil {
+		copy.MemInfoEx = &MemoryInfoExStat{}
+		*copy.MemInfoEx = *s.MemInfoEx
+	}
+	if s.IOStat != nil {
+		copy.IOStat = &IOCountersStat{}
+		*copy.IOStat = *s.IOStat
+	}
+	if s.IORateStat != nil {
+		copy.IORateStat = &IOCountersRateStat{}
+		*copy.IORateStat = *s.IORateStat
+	}
+	if s.CtxSwitches != nil {
+		copy.CtxSwitches = &NumCtxSwitchesStat{}
+		*copy.CtxSwitches = *s.CtxSwitches
```

Comment:
Why are we writing to the struct twice?

Reply:
we are initializing the pointer, then copying the data to the struct.